// @mcp-b/webmcp-ts-sdk/dist/index.js
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { Server } from "@modelcontextprotocol/sdk/server/index.js";
import { mergeCapabilities, mergeCapabilities as mergeCapabilities$1 } from "@modelcontextprotocol/sdk/shared/protocol.js";
import { ReadBuffer, serializeMessage } from "@modelcontextprotocol/sdk/shared/stdio.js";
import { CallToolRequestSchema, CallToolResultSchema, CreateMessageRequestSchema, CreateMessageResultSchema, ElicitRequestSchema, ElicitResultSchema, ErrorCode, GetPromptRequestSchema, GetPromptResultSchema, InitializeRequestSchema, InitializeResultSchema, JSONRPCMessageSchema, LATEST_PROTOCOL_VERSION, ListPromptsRequestSchema, ListPromptsResultSchema, ListResourcesRequestSchema, ListResourcesResultSchema, ListToolsRequestSchema, ListToolsResultSchema, LoggingLevelSchema, ReadResourceRequestSchema, ReadResourceResultSchema, ResourceListChangedNotificationSchema, SUPPORTED_PROTOCOL_VERSIONS, ToolListChangedNotificationSchema } from "@modelcontextprotocol/sdk/types.js";
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

//#region src/no-op-validator.ts
/**
* A no-op JSON Schema validator that always returns valid.
*
* Use this in browser environments where:
* - ajv causes "Error compiling schema" errors
* - Validation is already handled elsewhere (e.g., by Zod)
* - You need to avoid eval/Function constructor restrictions
*
* @example
* ```typescript
* import { BrowserMcpServer } from '@mcp-b/webmcp-ts-sdk';
* import { NoOpJsonSchemaValidator } from '@mcp-b/webmcp-ts-sdk/no-op-validator';
*
* const server = new BrowserMcpServer(
*   { name: 'my-server', version: '1.0.0' },
*   { jsonSchemaValidator: new NoOpJsonSchemaValidator() }
* );
* ```
*/
var NoOpJsonSchemaValidator = class {
	/**
	* Returns a validator function that always passes.
	* The input data is passed through unchanged.
	*
	* @param _schema - The JSON Schema (ignored)
	* @returns A validator function that always returns valid
	*/
	getValidator(_schema) {
		return (input) => ({
			valid: true,
			data: input,
			errorMessage: void 0
		});
	}
};

//#endregion
//#region src/browser-server.ts
/**
* Browser-optimized MCP Server for Web Model Context API (navigator.modelContext).
*
* This class extends McpServer to support dynamic registration of tools, resources,
* and prompts after transport connection, which is required for the Web Model Context API
* where items are registered via navigator.modelContext.provideContext() at any time.
*
* Key differences from base McpServer:
* - Pre-registers tool, resource, and prompt capabilities before connection
* - Allows items to be registered after transport is connected
* - Uses no-op JSON Schema validator to avoid ajv errors in browser environments
* - Designed for browser environments where items arrive asynchronously
*/
var BrowserMcpServer = class extends McpServer {
	constructor(serverInfo, options) {
		const enhancedOptions = {
			...options,
			capabilities: mergeCapabilities$1(options?.capabilities || {}, {
				tools: { listChanged: true },
				resources: { listChanged: true },
				prompts: { listChanged: true }
			}),
			jsonSchemaValidator: options?.jsonSchemaValidator ?? new NoOpJsonSchemaValidator()
		};
		super(serverInfo, enhancedOptions);
	}
	/**
	* Override connect to ensure request handlers are initialized
	* BEFORE the transport connection is established.
	*
	* This prevents the "Cannot register capabilities after connecting to transport"
	* error when items are registered dynamically after connection.
	*/
	async connect(transport) {
		this.setToolRequestHandlers();
		return super.connect(transport);
	}
};

//#endregion
export { BrowserMcpServer, BrowserMcpServer as McpServer, CallToolRequestSchema, CallToolResultSchema, Client, CreateMessageRequestSchema, CreateMessageResultSchema, ElicitRequestSchema, ElicitResultSchema, ErrorCode, GetPromptRequestSchema, GetPromptResultSchema, InitializeRequestSchema, InitializeResultSchema, JSONRPCMessageSchema, LATEST_PROTOCOL_VERSION, ListPromptsRequestSchema, ListPromptsResultSchema, ListResourcesRequestSchema, ListResourcesResultSchema, ListToolsRequestSchema, ListToolsResultSchema, LoggingLevelSchema, NoOpJsonSchemaValidator, ReadBuffer, ReadResourceRequestSchema, ReadResourceResultSchema, ResourceListChangedNotificationSchema, SUPPORTED_PROTOCOL_VERSIONS, Server, ToolListChangedNotificationSchema, mergeCapabilities, serializeMessage };
